src/frame.jl

```diff
@@ -0,0 +1,111 @@
+# Custom SVD solver for least squares problems
+function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
+    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
+    U,S,V = svd(Am)
+    S⁺ = inv.(S)[:]
+    S⁺[S⁺ .> 1/tol] .= 0

```

Comment:
I probably would have done something like the following:
```julia
U,σ,V = svd(Am)
filter!(>(tol), σ)
r = length(σ)
V[:,1:r] * (inv.(σ) .* (U[:,1:r]' * b))
```
But it doesn't matter too much.